src/cryptography/hazmat/backends/openssl/rsa.py

```diff
@@ -463,3 +498,9 @@ def verify(self, signature, data, padding, algorithm):
         return _rsa_sig_verify(
             self._backend, padding, algorithm, self, signature, data
         )
+
+    def recover(self, signature, padding, algorithm):
+        _check_not_prehashed(algorithm)
```

Comment:
We should add a quick test to trigger this

Reply:
Good point! Fixed.